### ISSUE TITLE
Predict Next Purchase Tutorial: evalml split_data and problem formation

### ISSUE THREAD

**Issue by flowersw** [alteryx/compose#197]
Hi, I have very much enjoyed seeing the progress that the alteryx team is making on these projects, very exciting and impressive stuff!

I was going through the [predict next purchase tutorial](https://render.githubusercontent.com/view/ipynb?color_mode=light&commit=3a83a05c1e3595a0d929b505dfecb273020780ac&enc_url=68747470733a2f2f7261772e67697468756275736572636f6e74656e742e636f6d2f616c74657279782f636f6d706f73652f336138336130356331653335393561306439323962353035646665636232373330323037383061632f646f63732f736f757263652f6578616d706c65732f707265646963745f6e6578745f70757263686173652e6970796e62&nwo=alteryx%2Fcompose&path=docs%2Fsource%2Fexamples%2Fpredict_next_purchase.ipynb&repository_id=163425829&repository_type=Repository#Splitting-the-Data) , and I ran into a couple of small issue, and also had a question about the modeling approach...

It looks like the EvalML API has been updated since this was created

Upon running 

```
y = fm.pop('bought_product')

splits = evalml.preprocessing.split_data(
    X=fm,
    y=y,
    test_size=0.2,
    random_state=0,
)

X_train, X_holdout, y_train, y_holdout = splits
```

I hit `TypeError: split_data() missing 1 required positional argument: 'problem_type'`

- adding `problem_type='binary'` as a kwarg param fixed that

And again, in running

```
automl = evalml.AutoMLSearch(
    problem_type='binary',
    objective='f1',
    random_state=0,
    allowed_model_families=['catboost', 'random_forest'],
    max_iterations=3,
)
```

I see `ValueError: Must specify training data as a 2d array using the X_train argument`

which, I assume should be 
```
automl = evalml.AutoMLSearch(
    X_train=X_train,
    y_train=y_train,
    problem_type='binary',
    objective='f1',
    random_state=0,
    allowed_model_families=['catboost', 'random_forest'],
    max_iterations=3,
)
```
Looks like the other tutorials need to be updated too.

I'm happy to make these changes, assuming there isn't a crazy docs build process or something. But, should this tutorial be added as some sort of automated test, and/or evalml be pinned in test_requirements.txt?

Now, about the modeling approach...I see that is being modeled as a simple 'binary' problem. This may be naive, but I wondered if it made any sense to model it as a time series classification problem, instead (perhaps even a case of https://github.com/alteryx/evalml/issues/1506)? Some of the customers in this data set are repeats, and thus show up in multiple rows, making those rows _not_ independent of each other, I think. While some customers only show up once, as a single row.

From the tutorial - user 6851 shows up once, and user 8167 shows up twice:
![Screen Shot 2021-01-10 at 1 16 38 AM](https://user-images.githubusercontent.com/12040615/104115850-bc960680-52e1-11eb-896e-05d2be4cce27.png)

The turbofan degredation example seems to be even more "time series" oriented.

Some sort of k-fold cross-validation and/or a test/train split that doesn't take those repeat customers is what I would be most worried about. Mostly, just curious what your team's thoughts are about this and if there's a clear way that y'all make this kind of distinction.

 Thanks

**Comment by jeff-hernandez** [comment]
@flowersw thanks for bringing up this issue! In the docs, the EvalML version is currently pinned to 0.16.0, so even though the doc tests run on each commit, they aren't testing against the latest API changes. We can resolve that by unpinning the EvalML version in the doc requirements and updating the tutorials to the latest API changes. Any contributions are helpful and greatly appreciated. The doc build process is easy and runs automatically on each commit. Would you be interested in submitting a pull request for this issue?

Support for time series classification problem is currently [under active development](https://github.com/alteryx/evalml/issues/1580), but I think it definitely makes sense to model this problem that way.

**Comment by flowersw** [comment]
I see. Yea, I'd be interested in submitting a PR. I would follow your advice as to whether to unpin evalml at 0.16, as you're probably better in the loop about how the projects will continue to integrate, etc. But, if the docs reference the different projects, I would think that would make sense. I can plan to make the pin >= 0.17, if that sounds good.

And as for the time series - good to know. Unless you can somehow make the claim that subsequent rows from the same entity (user, bot, etc.) are independent. And, I'm not sure all the ways the evalml is accounting for this. There's a  lot of subtelty, but, that makes sense to me too.

**Comment by jeff-hernandez** [comment]
Awesome, that'd be great! Yeah, `>= 0.17` sounds good.

Subsequent rows from the same entity use different data points from the series to create the label. Not exactly sure how independent that would make it. I think using a time series split vs k-fold cross-validation would make sense.

**Comment by flowersw** [comment]
I've got a PR in for the syntax changes. I can open a new issue to update the problem styles once the time series classification work is complete, unless y'all would like to handle that process a different way
